fe/fe-core/src/main/java/org/apache/doris/analysis/ArrayLiteral.java

```diff
@@ -134,6 +134,33 @@ protected void toThrift(TExprNode msg) {
         msg.setChildType(((ArrayType) type).getItemType().getPrimitiveType().toThrift());
     }
 
+    @Override
+    public int hashCode() {
+        int code =  31 * super.hashCode();
+        for (Expr c : children) {
+            code = code + c.hashCode();
+        }
+        return code;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof ArrayLiteral)) {
+            return false;
+        }
+        if (this == o) {
+            return true;
+        }
+
+        ArrayLiteral that = (ArrayLiteral) o;
+        for (int i = 0; i < children.size(); i++) {
```

Comment:
why not use `Objects.equals(children, that.children)` ?

Reply:
done